internal/junit/junit.go

```diff
@@ -10,15 +10,55 @@ const FileName = "junit.xml"
 
 // TestCase maps to <testcase> element
 type TestCase struct {
-	Name       string `xml:"name,attr"`
+	Name string `xml:"name,attr"`
+	// Assertions is the number of assertions in the test case.
 	Assertions string `xml:"assertions,attr,omitempty"`
-	Time       string `xml:"time,attr"`
-	Timestamp  string `xml:"timestamp,attr"`
-	ClassName  string `xml:"classname,attr"`
-	Status     string `xml:"status,attr,omitempty"`
-	SystemOut  string `xml:"system-out,omitempty"`
-	Error      string `xml:"error,omitempty"`
-	Failure    string `xml:"failure,omitempty"`
+	// Time in seconds it took to run the test case.
+	Time string `xml:"time,attr"`
+	// Timestamp as specified by ISO 8601 (2014-01-21T16:17:18).
+	// Timezone is optional.
+	Timestamp string `xml:"timestamp,attr"`
+	ClassName string `xml:"classname,attr"`
+	// Status indicates success or failure of the test. May be used instead of
+	// Error, Failure or Skipped or in addition to them.
+	Status    string   `xml:"status,attr,omitempty"`
+	File      string   `xml:"file,attr,omitempty"`
+	SystemErr string   `xml:"system-err,omitempty"`
+	SystemOut string   `xml:"system-out,omitempty"`
+	Error     *Error   `xml:"error,omitempty"`
+	Failure   *Failure `xml:"failure,omitempty"`
+	Skipped   *Skipped `xml:"skipped,omitempty"`
```

Comment:
If you're wondering why these are pointers, it's because you can have them as empty elements, like `<skipped />`. So doing an empty struct value check does not mean that the element is not present. Need a pointer to distinguish.